pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1090,12 +1090,31 @@ func (m *ManagerImpl) isDevicePluginResource(resource string) bool {
 	return false
 }
 
-// GetAllocatableDevices returns information about all the devices known to the manager
 func (m *ManagerImpl) GetAllocatableDevices() ResourceDeviceInstances {
 	m.mutex.Lock()
-	resp := m.allDevices.Clone()
+	allDevs := m.allDevices.Clone()
+	healthyDevs := cloneHealthDeviceData(m.healthyDevices)
```

Comment:
need not clone

Reply:
The alternative is to make the critical section longer, and I don't think it's a better approach.
This (`GetAllocatableDevices`) is expected to be called "rarely" (e.g. not in a hot loop).